packages/orbit-components/src/ChoiceTile/index.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from "react";
+
+import {
+  StyledChoiceTile,
+  StyledChoiceTileContent,
+  StyledChoiceTileDescription,
+  StyledChoiceTileHeading,
+  StyledChoiceTileLabel,
+  StyledInputWrapper,
+} from "./styled";
+import type { Props } from "./types";
+import Stack from "../Stack";
+import Text from "../Text";
+import useMediaQuery from "../hooks/useMediaQuery";
+import Radio from "../Radio";
+import Checkbox from "../Checkbox";
+
+const InputContent = ({ fullWidth, selected, label, type }: Props & { fullWidth?: boolean }) => {
+  return (
+    <StyledInputWrapper fullWidth={fullWidth}>
+      <StyledChoiceTileLabel selected={selected}>{label}</StyledChoiceTileLabel>
+      {type === "radio" ? (
+        <Radio readOnly tabIndex={-1} checked={selected} />
+      ) : (
+        <Checkbox readOnly tabIndex={-1} checked={selected} />
+      )}
+    </StyledInputWrapper>
+  );
+};
+
+const ChoiceTile = ({
+  children,
+  icon,
+  title,
+  description,
+  inline,
+  label,
+  type,
+  selected,
+  onClick,
+  dataTest,
+  id,
+}: Props) => {
+  const { isTablet } = useMediaQuery();
+  const onlyCustomContent = !icon && !title && !description && !!children;
+
+  return (
+    <StyledChoiceTile
+      selected={selected}
+      onClick={onClick}
+      tabIndex={0}
+      data-test={dataTest}
+      id={id}
+    >
+      <Stack inline spacing="medium" align="center">
+        {icon}
+        {title && (
```

Comment:
could you try to use a slightly different markup here? 

something like: 

```
 <Stack inline shrink spacing="medium">
        {icon}
        <Stack direction="column" spacing="XXSmall">
          {title && (
            <Stack inline spacing="XSmall" align="center">
              <StyledChoiceTileHeading>{title}</StyledChoiceTileHeading>
              {inline && isTablet && description && <Text type="secondary">{description}</Text>}
            </Stack>
          )}
          {(!isTablet || !inline) && description && <Text type="secondary">{description}</Text>}
        </Stack>
        {isTablet && <InputContent selected={selected} label={label} type={type} />}
      </Stack>
```

feel free to resolve it, if won't working because of some use case, but I will definitely try to place description under title in the same Stack, in order to prevent having custom text wrapper with margin offset

Reply:
It seems to work for title, description and input if `shrink` is applied to that new Stack and `InputContent` changes it `align-self` property to `start`. However, the Flag would need a wrapper, so that it is aligned to the top, with correct line-height and all. I think in the end it would be the same, removing one wrapper to add another 🤔

Reply:
For Flag, the wrapper will be smaller. Also for that case you can use only a div with `css` property ;) should be small and no need to have external styled wrapper for that

Reply:
I managed to be able to do it without any wrapper 👍 so I could remove the `StyledChoiceTileDescription` and adjust the icon positioning so it is inline with the title. Thanks for the suggestion!

Reply:
Great ;)